prov/verbs/src/verbs_info.c

```diff
@@ -1207,6 +1209,32 @@ int fi_ibv_init_info(const struct fi_info **all_infos)
 	ep_type[2] = &verbs_dgram_domain;
 
 	if (!fi_ibv_gl_data.fork_unsafe) {
+		/* check to see if hugepages can be used */
+		buffer_size = ofi_get_hugepage_size();
+		if (buffer_size > 0) {
+			/* attempt to map hugepage if valid size found */
+			buffer = mmap(NULL, buffer_size,
+				PROT_READ | PROT_WRITE,
+				MAP_PRIVATE | MAP_ANONYMOUS | MAP_HUGETLB,
+				-1, 0);
```

Comment:
Use ofi_alloc_hugepage_buf() and ofi_free_hugepage_buf().  We should probably create a new abstraction for this.  E.g:
```
int ofi_hugepage_enabled()
{
    size = ofi_get_hugepage_size();
    if (!size)
        return 0;
    ret = ofi_alloc_hugepage_buf(&ptr, size);
    if (ret)
        return 0;
    ofi_free_hugepage_buf(ptr, size);
    return 1;
}
The ofi_xxx_hugepage_xxx calls are scattered in the various osd.h files.  We should also define the function prototypes in maybe ofi_mem.h, so they are easier to find.

The hugepage support is also needed by the userfaultfd work (#4946).

Reply:
I've added a separate commit for the function. It looks like we won't need it, but we could cherry pick it out and keep it for later. 

@shefty thoughts?

Reply:
Yes - regardless of what we decide, I think we should pull in that change.  Thanks!